src/reconcile/rec_write.c

```diff
@@ -292,7 +292,8 @@ __reconcile(WT_SESSION_IMPL *session, WT_REF *ref, WT_SALVAGE_COOKIE *salvage, u
      */
     if (ret == 0 && !(btree->evict_disabled > 0 || !F_ISSET(btree->dhandle, WT_DHANDLE_OPEN)) &&
       F_ISSET(r, WT_REC_EVICT) && !WT_PAGE_IS_INTERNAL(r->page) && r->multi_next == 1 &&
-      F_ISSET(r, WT_REC_CALL_URGENT) && !r->update_used && r->cache_write_restore) {
+      F_ISSET(r, WT_REC_CALL_URGENT) && !r->update_used &&
+      (r->cache_write_restore_invisible || !r->cache_discard_update_chain)) {
```

Comment:
I think `!r->cache_discard_update_chain` should || with` !r->update_used`?

Reply:
no, it should be the current way. if r->update_used is true, we are making progress, no need to check the following.

Reply:
I think it should be !r->update_used && r->cache_write_restore_invisible &&
      !r->cache_discard_update_chain